marimo/_messaging/ops.py

```diff
@@ -539,9 +539,9 @@ def _stringify(self, value: object) -> str:
             if table_manager is not None:
                 return str(table_manager)
             else:
+                # TODO(akshayka): str(value) can be extremely expensive
+                # for some objects; find a better solution.
                 return str(value)[:50]
```

Comment:
Some low hanging fruit could be to see if narwhals can digest the data (implements array interface if that's the case), and return a stringed slice based on that

Reply:
we do use narwhals for the dataframe stuff. it doesn't handle just arrays. but I've written something custom to handle those: https://github.com/marimo-team/marimo/pull/3635